caffe2/core/tensor.cc

```diff
@@ -201,9 +201,16 @@ void Tensor::enforce_invariants() {
   if (impl_.get() == nullptr) {
     throw std::runtime_error("TensorImpl with nullptr is not supported");
   }
-  CAFFE_ENFORCE(
-      !impl_->is_variable(),
-      "Caffe2 tensor wrapper doesn't support autograd variables");
+  // TODO: only check `!impl_->requires_grad()` after Variable and Tensor are merged
+#ifndef CAFFE2_FB_LIMITED_MOBILE_CAPABILITY
+    CAFFE_ENFORCE(
+      !impl_->is_variable() || (impl_->is_variable() && !(impl_->requires_grad() && torch::autograd::GradMode::is_enabled())),
```

Comment:
you don't need to check is_variable again -- if it got to the or, you know it's a variable.